spec/models/user_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+
+  context 'field validations' do
+    it { should validate_presence_of(:first_name) }
+    it { should validate_presence_of(:last_name) }
+  end
+
+  context 'testing Associations' do
+    it { should have_many(:posts).dependent(:destroy)}
+    it { should have_many(:comments).dependent(:destroy) }
+  end
+
+  context 'friend_requests' do
+    it { should have_many(:friends_requested).through(:sent_requests) }
+    it { should have_many(:friends_requesting).through(:received_requests)}
+  end
+
+  context 'friendship' do
+    it { should have_many(:direct_friends).through(:direct_friendships) }
+    it { should have_many(:inverse_friends).through(:inverse_friendships)}
+  end
+
+  describe 'public instance methods' do
+    let(:user) { create(:user) }
+    context 'respond to its methods' do
```

Comment:
This may not be the best way to write tests for your instance methods. For instance it's obvious that `user` will respond to `friends` method. But are we sure about what that method will return? So simply testing if your user object will respond to these methods isn't enough to catch any problem.